### PR TITLE
Update RUST dependencies and audit workflow

### PR DESCRIPTION
**Pull request by awharn**
Updates the daemon dependencies and the RUST workflow to find additional issues, like yanked crates.
Workflow result before: https://github.com/zowe/zowe-cli/actions/runs/4426805429
Workflow result after: https://github.com/zowe/zowe-cli/actions/runs/4427115394/jobs/7764266287

### CODE REVIEW

**Review by zFernand0 — APPROVED** [review]
Thanks for updating `bumpalo` 🥳 
And also for updating the workflow 😋 

LGTM! 😋

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Will approve once additional warnings that appear with `--deny warnings` flag have been addressed 🙂

**Review by traeok — APPROVED** [review]
LGTM, thanks @awharn !

**Review by gejohnston — APPROVED** [review]

File: .github/workflows/rust-cli-publish.yml [hunk 1]
```diff
@@ -110,39 +103,36 @@ jobs:
...
-    - run: echo ${{ steps.get_upload_url.outputs.upload_url }}
+    - name: Get version
+      id: get-version
+      working-directory: zowex
+      run: echo "ZOWEX_VERSION=$(cargo metadata --no-deps | jq -r .packages[0].version)" >> $GITHUB_OUTPUT
 
     - name: Create Archive
       run: |
         cd target/release
-        gtar -cvzf zowe.tgz zowe
```
Comment (t1m0thyj) [inline .github/workflows/rust-cli-publish.yml:135]: Previously we had to use `gtar` to work around a bug where `tar` on MacOS included an extra file in the archive: https://github.com/zowe/zowe-cli/pull/1311

This seems to no longer be a problem, so I'm cool with using `tar` 🙂

File: .github/workflows/rust-cli.yml
Comment (t1m0thyj) [inline .github/workflows/rust-cli.yml:1]: Noticed that we have duplicate runs of this workflow on `push` and `pull_request` events. Do we want to add the same condition we have in other workflows to skip duplicate runs?
`if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository) && !contains(github.event.head_commit.message, '[ci skip]')`

**Review by t1m0thyj — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**